Mage.Sets/src/mage/cards/s/SmugglersShare.java

```diff
@@ -0,0 +1,135 @@
+        ability.addEffect((new CreateTokenEffect(new TreasureToken(), SmugglersShareTreasureValue.instance)).concatBy(", then"));
+        ability.addWatcher(new CardsAmountDrawnThisTurnWatcher());
+        ability.addWatcher(new PermanentsEnteredBattlefieldWatcher());
+        this.addAbility(ability);
+    }
+
+    private SmugglersShare(final SmugglersShare card) {
+        super(card);
+    }
+
+    @Override
+    public SmugglersShare copy() {
+        return new SmugglersShare(this);
+    }
+}
+
+enum SmugglersShareDrawValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability source, Effect effect) {
+        Player controller = game.getPlayer(source.getControllerId());
+        CardsAmountDrawnThisTurnWatcher watcher = game.getState().getWatcher(CardsAmountDrawnThisTurnWatcher.class);
+        int drawCount = 0;
+        if (watcher != null) {
+            for (UUID playerId : game.getOpponents(controller.getId())) {
+                Player opponent = game.getPlayer(playerId);
+                if (opponent != null) {
+                    if (watcher.getAmountCardsDrawn(playerId) >= 2) {
+                        drawCount++;
+                    }
+                }
+            }
+        }
+        return drawCount;
+    }
+
+    @Override
+    public SmugglersShareDrawValue copy() {
+        return instance;
+    }
+
+    @Override
+    public String getMessage() {
+        return "opponent who drew two or more cards this turn";
+    }
+
+    @Override
+    public String toString() {
+        return "1";
+    }
+}
+
+enum SmugglersShareTreasureValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability source, Effect effect) {
+        Player controller = game.getPlayer(source.getControllerId());
+        PermanentsEnteredBattlefieldWatcher watcher = game.getState().getWatcher(PermanentsEnteredBattlefieldWatcher.class);
+        int treasureCount = 0;
+        if (watcher != null) {
+            for (UUID opponentId : game.getOpponents(controller.getId())) {
+                Player opponent = game.getPlayer(opponentId);
+                if (opponent != null) {
+                    List<Permanent> enteredPermanents = watcher.getThisTurnEnteringPermanents(opponentId);
+                    if (enteredPermanents != null) {
+                        int enteredLandCount = 0;
+                        for (Permanent permanent : enteredPermanents) {
+                            if (permanent.isLand(game)) {
+                                enteredLandCount++;
+                            }
+                        }
+                        if (enteredLandCount >= 2) {
+                            treasureCount++;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Can you please flatten these if statements.

e.g. Instead of
```Java
if (watcher != null) {
   // Do something
}
```
Use
```Java
if (watcher == null) {
    return 0;
}
```

Or continue through the loop.
It makes the code more readable by not having highly nested statements

Reply:
Sure